src/lib/gossip_net/gossip_net.ml

```diff
@@ -132,6 +132,15 @@ module Make (Message : Message_intf) : S with type msg := Message.msg = struct
     | Ok conn ->
         Versioned_rpc.Connection_with_menu.create conn
 
+  (* remove peer from set of peers and peers_by_ip *)
```

Comment:
Please add a comment explaining what's wrong with this approach: A) if a node disconnects from the network and then reconnects we may not add it to our peer set again since the Kademlia process doesn't know it was gone in the first place, B) Kademlia will continue to propogate the connection info for these disconnected peers to others in peer discovery.